v2/bigtable-common/src/main/java/com/google/cloud/teleport/v2/bigtable/options/BigtableCommonOptions.java

```diff
@@ -137,4 +139,147 @@ interface WriteOptions extends BigtableCommonOptions {
 
     void setBigtableBulkWriteMaxRequestSizeBytes(Integer value);
   }
+
+  interface ReadOptions extends BigtableCommonOptions {
+
+    @TemplateParameter.Text(
+        order = 1,
+        regexes = {"[a-z][a-z0-9\\-]+[a-z0-9]"},
+        description = "Source Bigtable Instance ID",
+        helpText = "The ID of the Cloud Bigtable instance that contains the table")
+    @Validation.Required
+    String getBigtableReadInstanceId();
+
+    void setBigtableReadInstanceId(String value);
+
+    @TemplateParameter.Text(
+        order = 2,
+        description = "Source Cloud Bigtable table ID",
+        helpText = "The Cloud Bigtable table to read from.")
+    @Validation.Required
+    String getBigtableReadTableId();
+
+    void setBigtableReadTableId(String value);
+
+    @TemplateParameter.ProjectId(
+        order = 3,
+        optional = true,
+        description = "Source Cloud Bigtable Project ID",
+        helpText =
+            "Project to read Cloud Bigtable data from. The default for this parameter is the "
+                + "project where the Dataflow pipeline is running.")
+    @Default.String("")
+    String getBigtableReadProjectId();
+
+    void setBigtableReadProjectId(String projectId);
+
+    @TemplateParameter.Text(
+        order = 4,
+        optional = true,
+        regexes = {"[a-z][a-z0-9\\-_]+[a-z0-9]"},
+        description = "Bigtable App Profile",
+        helpText =
+            "Bigtable App Profile to use for reads. The default for this parameter "
+                + "is the Bigtable instance's default app profile")
+    @Default.String("default")
+    String getBigtableReadAppProfile();
+
+    void setBigtableReadAppProfile(String value);
+  }
+
+  interface ReadChangeStreamOptions extends BigtableCommonOptions.ReadOptions {
+
+    @TemplateParameter.Text(
+        order = 1,
+        optional = true,
+        description = "Cloud Bigtable change streams metadata instance ID",
+        helpText =
+            "The Cloud Bigtable instance to use for the change streams connector metadata table.")
+    @Default.String("")
+    String getBigtableChangeStreamMetadataInstanceId();
+
+    void setBigtableChangeStreamMetadataInstanceId(String value);
+
+    @TemplateParameter.Text(
+        order = 2,
+        optional = true,
+        description = "Cloud Bigtable change streams metadata table ID",
+        helpText =
+            "The Cloud Bigtable change streams connector metadata table ID to use. If not "
+                + "provided, a Cloud Bigtable change streams connector metadata table will automatically be "
+                + "created during the pipeline flow. This parameter must be provided when updating an "
+                + "existing pipeline and should not be provided otherwise.")
```

Comment:
the param doesn't have to be provided, you just have to use the same metadata table for each pipeline run. If they both use the default it's fine

Reply:
Agree